src/Morpho.sol

```diff
@@ -194,8 +194,8 @@ contract Morpho is IMorpho {
         Id id = marketParams.id();
         require(market[id].lastUpdate != 0, ErrorsLib.MARKET_NOT_CREATED);
         require(UtilsLib.exactlyOneZero(assets, shares), ErrorsLib.INCONSISTENT_INPUT);
-        // No need to verify that onBehalf != address(0) thanks to the authorization check.
         require(receiver != address(0), ErrorsLib.ZERO_ADDRESS);
+        // No need to verify that onBehalf != address(0) thanks to the following authorization check.
         require(_isSenderAuthorized(onBehalf), ErrorsLib.UNAUTHORIZED);
```

Comment:
to me we shouldn't change this, because the comment is in the place where you expect the check (based on the functions where the check is)

Reply:
It works only for those who know very well the code like us. Not for others. So I'm against reverting the change